src/store/reader.rs

```diff
@@ -74,24 +130,30 @@ impl StoreReader {
         self.data.slice(checkpoint.byte_range.clone()).read_bytes()
     }
 
+    /// Loads and decompresses a block.
+    ///
+    /// Advanced API. In most cases use [get](Self::get).
     fn read_block(&self, checkpoint: &Checkpoint) -> io::Result<Block> {
-        if let Some(block) = self.cache.lock().unwrap().get(&checkpoint.byte_range.start) {
-            self.cache_hits.fetch_add(1, Ordering::SeqCst);
-            return Ok(block.clone());
+        if let Some(block) = self.cache.get_from_cache(checkpoint.byte_range.start) {
+            return Ok(block);
         }
 
-        self.cache_misses.fetch_add(1, Ordering::SeqCst);
-
         let compressed_block = self.compressed_block(checkpoint)?;
+
+        self.decompress_block(compressed_block, checkpoint.byte_range.start)
+    }
+
+    fn decompress_block(
```

Comment:
see comment above. This applies to the other call site too.

`decompress_block` might be better as function (= not a method) after that change.

Or alternatively, it could be helper method with a blanket impl on the compressor side.

Reply:
Moved to compressor